site/content/docs/development/architecture.md

```diff
@@ -0,0 +1,144 @@
+  * The path field depends upon the specific template kind.
+
+![Template](../img/template.jpg)
+
+Templates are typed by the output they produce.
+
+| Output      | Template |
+| ----------- | ----------- |
+| Config | [ClusterConfigTemplate](reference/#clusterconfigtemplate) |
+| Image | [ClusterImageTemplate](reference/#clusterimagetemplate) |
+| Source | [ClusterSourceTemplate](reference/#clustersourcetemplate) |
+| Deployment | [ClusterDeploymentTemplate](reference/#clusterdeploymenttemplate) |
+| | [ClusterTemplate](reference/#clustertemplate) |
+
+### Owners
+
+Owners represent the workload or deliverable, which in many cases refer to a single application's source or image 
+location.
+
+Owners are the developer provided configuration which cause a blueprint to be reconciled into resources.
+
+They consist of:
+* **Labels**: blueprints will select based on the labels of an owner, see [selectors](#selectors) 
+* **Params**: parameters supplied to the blueprint, see [Parameter Hierarchy](#parameter-hierarchy)
+* **Source**: The source reference for the input to the Supply Chain or Delivery Blueprints,
+see [Workload](reference.md/#workload) and [Deliverable](reference.md/#deliverable)
+
+![Owner](../img/owner.jpg)
+
+| Owner      | Blueprint |
+| ----------- | ----------- |
+| Workload | ClusterSupplyChain |
+| Deliverable | ClusterDelivery |
+
+## Theory of operation
+
+Given an owner that matches a blueprint, Cartographer reconciles the resources referenced by the blueprint.
+The resources are only created when the inputs are satisfied, and a resource is only updated when it's inputs change.
+This results in a system where a new result from one resource can cause other resources to change.
+
+![Generic Blueprint](../img/generic.jpg)
+<!-- https://miro.com/app/board/uXjVOeb8u5o=/ -->
+
+Although Cartographer is not a 'runner of things', a resource can be something as simple as a Job or a CI pipeline.
+
+However, one advantage of Cartographer's design is that resources that self-mutate can cause downstream change.
+
+For example, a Build resource that discovers new base OCI images. If it rebuilds your image, then Cartographer will see 
+this new image and update downstream resources.
+
+When Cartographer reconciles an owner, each resource in the matching blueprint is applied:
+
+1. **Generate Inputs**: Using the **blueprint resource's** `inputs` as a reference, select outputs from previously applied **Kubernetes resources**
+2. **Generate Params**: Using the [Parameter Hierarchy](architecture.md#parameter-hierarchy), generate parameter values   
+3. **Generate and apply resource spec**: Apply the result of interpolating `spec.template` (or `spec.ytt`), **inputs**, **params** and the **owner spec**. 
+4. **Retrieve Output**: Store the output from the applied resource. The output to use is specified in the **template output path**.  
+
+![Realize](../img/realize.jpg)
```

Comment:
Dependency arrow always throw me off. That's a me thing.

I think the dependency arrow on the top output is facing the wrong way.

I'm not sure that the bottom output should have a dependency arrow.

Reply:
yeah because we needed a data flow arrow at the bottom, we used one at the top, there's no great convention for it other than to create a key and use a decoration.

Reply:
I really wanted a 'data bucket' for in memory 'outputs', which we could then simply 'depend' upon, however the diagram gets very confusing that way too.

Reply:
If you want to try your hand at an alternative, this is the board: https://miro.com/app/board/uXjVOeb8u5o=/